File: solutions/javascript/hamming/1/hamming.js
var Hamming = function() {

}

Hamming.prototype.compute = function(strandA, strandB) {
  // Check that strands are equal in length
  if (strandA.length != strandB.length) {
    throw new Error('DNA strands must be of equal length.')
  }
  else {
    hamming = 0
    for(var i = 0; i < strandA.length; i++) {
      if(strandA[i] !== strandB[i]) {
        hamming ++
      }
    }
  }
  return hamming
}

module.exports = Hamming

File: solutions/go/hamming/1/hamming.go
package hamming

import (
	"errors"
)

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("sequences must be of equal length")
	}

	distance := 0

	for i, _ := range(a) {
		if a[i] != b[i] {
			distance ++
		}
	}

	return distance, nil
}
